Index: layout/1/planwatch.inc
===================================================================
RCS file: /cvs/planworld/layout/1/planwatch.inc,v
retrieving revision 1.18
diff -u -r1.18 planwatch.inc
--- layout/1/planwatch.inc	23 Aug 2002 21:27:31 -0000	1.18
+++ layout/1/planwatch.inc	19 Feb 2003 02:23:55 -0000
@@ -42,6 +42,8 @@
   foreach ($pw as $name => $group) {
     echo "<strong><u>{$name}</u></strong><br />\n";
     foreach ($group as $u => $entry) {
+      if (isset($entry[4]) && $entry[4])
+      	echo "<a href=\"/message.jsp?id=" . stripslashes($u) . "\" title=\"pick up message(s)\"><img src=\"/images/mail.gif\" alt=\"message\" border=\"0\" /></a>";
       if ($entry[1] > $entry[2])
 	echo "<span class='new'>NEW</span> ";
       echo "<a class=\"planwatch\" href=\"" . PW_URL_INDEX . "?id=" . stripslashes($u) . "\" title=\"Finger " . stripslashes($u) . "\">" . stripslashes($u) . "</a> ";
Index: layout/1/who.inc
===================================================================
RCS file: /cvs/planworld/layout/1/who.inc,v
retrieving revision 1.9
diff -u -r1.9 who.inc
--- layout/1/who.inc	30 May 2002 15:40:07 -0000	1.9
+++ layout/1/who.inc	19 Feb 2003 02:23:55 -0000
@@ -48,7 +48,7 @@
 </tr>
 <?php } ?>
 <tr>
-<td colspan="<?php echo ($_user->isAdmin()) ? '4' : '3'; ?>" class="entry">Users idle for more than 10 minutes are considered inactive.</td>
+<td colspan="<?php echo ($_user->isAdmin()) ? '4' : '3'; ?>" class="entry">Users idle for more than 3 minutes are considered inactive.</td>
 </tr>
 </table>
 </td>
@@ -56,4 +56,4 @@
 </table>
 <?php
 } // if (is_array($who) && !empty($who)) 
-?>
\ No newline at end of file
+?>
Index: lib/Planwatch.php
===================================================================
RCS file: /cvs/planworld/lib/Planwatch.php,v
retrieving revision 1.9.2.3
diff -u -r1.9.2.3 Planwatch.php
--- lib/Planwatch.php	20 Sep 2002 02:26:20 -0000	1.9.2.3
+++ lib/Planwatch.php	19 Feb 2003 02:23:55 -0000
@@ -34,7 +34,7 @@
    */
   function load ($sort = null) {
     /* assemble the query */
-    $query = "SELECT u.username, u.id, p.last_view, u.last_update, g.name AS name, g.gid AS gid, g.uid AS owner FROM pw_groups AS g, planwatch AS p, users AS u WHERE p.uid=" . $this->user->getUserID() . " AND p.w_uid=u.id AND g.gid=p.gid ORDER BY g.pos, g.name,";
+    $query = "SELECT u.username, u.id, p.last_view, u.last_update, g.name AS name, g.gid AS gid, g.uid AS owner, m.seen AS hasmessage FROM pw_groups AS g, planwatch AS p, users AS u LEFT JOIN message AS m ON (m.uid=p.w_uid AND m.to_uid=p.uid) WHERE p.uid=" . $this->user->getUserID() . " AND p.w_uid=u.id AND g.gid=p.gid ORDER BY g.pos, g.name,";
 
     if (!isset($sort)) {
       $sort = $this->user->getWatchOrder();
@@ -68,7 +68,8 @@
 	$this->planwatch[$user] = array((int) $row['id'],
 					(int) $row['last_update'],
 					(int) $row['last_view'],
-					false);
+					false,
+					((isset($row['hasmessage']) && $row['hasmessage'] == 0) ? true : false));
 	/* create a pointer to this entry within the appropriate group */
 	$this->groupData[$group][$user] = &$this->planwatch[$user];
 
@@ -89,6 +90,25 @@
         $this->planwatch[$username]['count'] = 2;
       }
       $this->groupData['Snoop'][$username] = &$this->planwatch[$username];
+    }
+
+    /* get send group */
+    $query = "SELECT u.username, u.id, u.last_update FROM users AS u INNER JOIN message ON u.id=message.uid LEFT JOIN online ON message.uid=online.uid WHERE message.to_uid=" . $this->user->getUserID() . " AND message.seen=0 ORDER BY username";
+
+    /* execute the query */
+    $result = $this->dbh->query($query);
+    if (isset($result) && !DB::isError($result)) {
+      while ($row = $result->fetchrow(DB_FETCHMODE_ASSOC)) {
+	$username = $row['username'];
+	if (!isset($this->planwatch[$username])) {
+	  $this->planwatch[$username] = array($row['id'], $row['last_update'], 9999999999);
+	} else if (isset($this->groupData['Snoop'][$username])) {
+	  $this->planwatch[$username]['count'] = 3;
+	} else {
+	  $this->planwatch[$username]['count'] = 2;
+	}
+	$this->groupData['Send'][$username] = &$this->planwatch[$username];
+      }
     }
 
     $this->changed = false;
Index: lib/User.php
===================================================================
RCS file: /cvs/planworld/lib/User.php,v
retrieving revision 1.20.2.5
diff -u -r1.20.2.5 User.php
--- lib/User.php	9 Oct 2002 02:31:24 -0000	1.20.2.5
+++ lib/User.php	19 Feb 2003 02:23:55 -0000
@@ -600,7 +600,7 @@
       $out .= "<tt>Login name: {$this->username}";
       if ($this->isUser() && $this->isSharedFor($user))
 	$out .= " (<a href=\"" . PW_URL_INDEX . "?id=edit_plan;u={$this->username}\">edit</a>)";
-      $out .= "<br />\n";
+      $out .= " (<a href=\"/message.jsp?id=" . $this->username . "\" title=\"send to " . $this->username . "\">send</a>)<br />\n";
 
       /* user doesn't exist */
       if (!$this->isUser() || ($this->lastLogin == 0 && $this->lastUpdate == 0)) {
